core/views/files.py

```diff
@@ -55,9 +61,14 @@ def send_file(request, file_id, file_class=SithFile, file_attr="file"):
             ).exists()
         )
     ):
-        raise PermissionDenied
+        return forbidden(request, _("You are not allowed to view this file"))
     name = f.__getattribute__(file_attr).name
     filepath = os.path.join(settings.MEDIA_ROOT, name)
+
+    # check if file exists on disk
+    if not os.path.exists(filepath.encode("utf-8")):
+        return not_found(request, _("File not found"))
```

Comment:
```suggestion
        raise Http404
```
Faut aussi faire l'import, si c'est pas fait.

Reply:
Parce que c'est le fichier qui n'est pas trouvé, pas la page, donc autant retourner la page 404 avec l'erreur que juste l'erreur 404. De plus, c'est à mon sens plus propre de faire un return qu'un raise.

Reply:
Pas dans le sens de Django, parce que :

- 404, ça n'indique pas une page non trouvée, mais une ressource de manière générale, donc ici le 404 est pertinent.
- Tu peux vérifier, le message que tu précises dans l'appel de ta fonction n’apparaît pas sur la page 404.
- Quand une erreur Http404 est levée, Django prend automatiquement en charge le routage : quand le projet est en mode DEBUG, on obtient une page avec la pile d'appels et une description de l'erreur et quand il n'est pas en mode DEBUG, ça appelle la vue not_found(). Bref, ça fait exactement ce que tu fais, mais en mieux, parce que plus facile à débuguer et plus idiomatique.

Tu peux vérifier, il n'y a aucun appel en clair à la vue not_found dans le projet, c'est que du raise Http404.